From 1d45a1987073ad1e96a490993cefdfec7088c1c3 Mon Sep 17 00:00:00 2001
From: Rui Loura <rui@joyent.com>
Date: Thu, 13 Dec 2018 19:19:15 +0000
Subject: [PATCH] TRITON-1033 adminIpFromSysinfo should check for missing NIC
 Reviewed by: Robert Bogart <Robert.bogart@joyent.com> Approved by: Robert
 Bogart <Robert.bogart@joyent.com>

---
 lib/index.js | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lib/index.js b/lib/index.js
index f9a8686..f0ae562 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -83,7 +83,11 @@ function adminIpFromSysinfo(sysinfo) {
 
     var nic = adminNicFromSysinfo(sysinfo);
 
-    return nic.ip4addr;
+    if (nic) {
+        return nic.ip4addr;
+    }
+
+    return null;
 }
 
 // ---- From VM Metadata
-- 
2.21.0

